pkg/providers/tinkerbell/config/template-cp.yaml

```
@@ -280,9 +303,11 @@ spec:
         path: "/etc/containerd/config_append.toml"
 {{- end }}
 {{- end }}
-{{- if and .registryMirrorMap (ne .format "bottlerocket") }}
+{{- if and (or .proxyConfig .registryMirrorMap) (ne .format "bottlerocket") }}
     preKubeadmCommands:
+{{- if and .registryMirrorMap (ne .format "bottlerocket") }}
```

nit: Line already 306 ensures that the format isn't bottlerocket so we don't need to check again
```suggestion
{{- if .registryMirrorMap }}
```